### PR TITLE
Feature/Export custom rule to terraform file.

### PR DESCRIPTION
**Pull request by madhusarma**
### Description of feature:

This PR adds the new command "export", to export the custom rule to terraform file. 

Here is a sample command to export the rule:

>>**rdk export  << rule-name >>  -f terraform -v 0.12**

It generates the three .tf files in the current directory.

- << rule-name >>_custom_rule.tf : terraform script, that upload the rule to S3, deploy the lambda, create the aws config rule and required IAM roles/policies.
- << rule-name >>_variables.tf: Varabiles definitions.
- << rule-name >>.tfvars.json: Variable values.

Then use the below terraform command to deploy the rule.
**>>terraform  apply  -var-file="<< rule-name >>.tfvars.json"  --var source_bucket=<< existing s3 bucket name >>**

Use the below command to get help of this new command.

> **rdk export -h**

> usage: rdk export [-h] [-s RULESETS] [--all] [--lambda-layers LAMBDA_LAYERS]
>                   [--lambda-subnets LAMBDA_SUBNETS]
>                   [--lambda-security-groups LAMBDA_SECURITY_GROUPS]
>                   [--lambda-role-arn LAMBDA_ROLE_ARN]
>                   [--rdklib-layer-arn RDKLIB_LAYER_ARN] -v {0.11,0.12} -f
>                   {terraform}
>                   [<rulename> [<rulename> ...]]
> 
> Used to export the Config Rule to terraform file.
> 
> positional arguments:
>   <rulename>            Rule name(s) to export to a file.
> 
> optional arguments:
>   -h, --help            show this help message and exit
>   -s RULESETS, --rulesets RULESETS
>                         comma-delimited list of RuleSet names
>   --all, -a             All rules in the working directory will be deployed.
>   --lambda-layers LAMBDA_LAYERS
>                         [optional] Comma-separated list of Lambda Layer ARNs
>                         to deploy with your Lambda function(s).
>   --lambda-subnets LAMBDA_SUBNETS
>                         [optional] Comma-separated list of Subnets to deploy
>                         your Lambda function(s).
>   --lambda-security-groups LAMBDA_SECURITY_GROUPS
>                         [optional] Comma-separated list of Security Groups to
>                         deploy with your Lambda function(s).
>   --lambda-role-arn LAMBDA_ROLE_ARN
>                         [optional] Assign existing iam role to lambda
>                         functions. If omitted, new lambda role will be
>                         created.
>   --rdklib-layer-arn RDKLIB_LAYER_ARN
>                         [optional] Lambda Layer ARN that contains the desired
>                         rdklib. Note that Lambda Layers are region-specific.
>   -v {0.11,0.12}, --version {0.11,0.12}
>                         Terraform version
>   -f {terraform}, --format {terraform}
>                         Export Format
>

### CODE REVIEW

File: rdk/rdk.py [hunk 1]
```diff
@@ -1259,6 +1280,109 @@ def deploy(self):
...
+            s3_dst = self.__package_function_code(rule_name, rule_params)
+
+            layers = []
+            rdk_lib_version = "0"
+            if 'SourceRuntime' in rule_params:
+                if rule_params['SourceRuntime'] == "python3.6-lib":
+                    if self.args.rdklib_layer_arn:
+                        layers.append(self.args.rdklib_layer_arn)
+                    else:
+                        rdk_lib_version = RDKLIB_LAYER_VERSION[my_session.region_name]
```
Comment (rickychau2780) [inline rdk/rdk.py:1335]: NameError: name 'my_session' is not defined

Comment (madhusarma) [reply rdk/rdk.py:1337]: > Thanks for contribution!! Please also add documentation in /doc/reference for new commands

Updated the documentation.

Comment (madhusarma) [reply rdk/rdk.py:1337]: > NameError: name 'my_session' is not defined

Resolved this error and updated the PR. Thanks for the quick reply.

**Comment by madhusarma** [comment]
Updated the PR, please review.

**Comment by rickychau2780** [comment]
Thank you. looks good to me.

**Pull request merged**